src/form/input/typeahead/TypeaheadInput.tsx

```diff
@@ -1,55 +1,31 @@
 import React, {useEffect} from "react";
 import classNames from "classnames";
 
-import Input, {InputTypes} from "../Input";
+import Input, {InputProps, InputTypes} from "../Input";
 import useDebounce from "../../../core/utils/hooks/debounce";
 
-export interface TypeaheadInputProps {
+export type TypeaheadInputProps = Omit<
+  InputProps,
+  "onChange" | "type" | "customClassName" | "value"
```

Comment:
I think we don't need to omit `customClassName` and `value` as they have the same type in InputProps as well

Reply:
Yes, you are right. Just destructuring is enough.